planetscale/imports.go

```diff
@@ -0,0 +1,321 @@
+		return "failed to detach external database from  PlanetScale database"
+	}
+
+	return "all done"
+}
+
+type DataImport struct {
+	ID          string `json:"id"`
+	ImportState DataImportState
+	State       string    `json:"state"`
+	Errors      string    `json:"import_check_errors"`
+	StartedAt   time.Time `json:"started_at"`
+	FinishedAt  time.Time `json:"finished_at"`
+	DeletedAt   time.Time `json:"deleted_at"`
+}
+
+func (di *DataImport) ParseState() {
+	switch di.State {
+	case "prepare_data_copy_pending":
+		di.ImportState = DataImportPreparingDataCopy
+	case "prepare_data_copy_error":
+		di.ImportState = DataImportPreparingDataCopyFailed
+	case "data_copy_pending":
+		di.ImportState = DataImportCopyingData
+	case "data_copy_error":
+		di.ImportState = DataImportCopyingDataFailed
+	case "switch_traffic_workflow_pending":
+		di.ImportState = DataImportSwitchTrafficPending
+	case "switch_traffic_workflow_running":
+		di.ImportState = DataImportSwitchTrafficRunning
+	case "switch_traffic_workflow_error":
+		di.ImportState = DataImportSwitchTrafficError
+	case "reverse_traffic_workflow_running":
+		di.ImportState = DataImportReverseTrafficRunning
+	case "reverse_traffic_workflow_error":
+		di.ImportState = DataImportReverseTrafficError
+	case "cleanup_workflow_pending":
+		di.ImportState = DataImportSwitchTrafficCompleted
+	case "cleanup_workflow_running":
+		di.ImportState = DataImportDetachExternalDatabaseRunning
+	case "cleanup_workflow_error":
+		di.ImportState = DataImportDetachExternalDatabaseError
+	default:
+		di.ImportState = DataImportReady
+	}
+}
+
+type TestDataImportSourceRequest struct {
+	Organization string           `json:"organization"`
+	Database     string           `json:"name"`
+	Source       DataImportSource `json:"connection"`
+}
+
+// DataSourceIncompatibilityError represents an error that occurs when the
+// source schema in an external database server is incompatible with PlanetScale.
+type DataSourceIncompatibilityError struct {
+	LintError        string `json:"lint_error"`
+	Keyspace         string `json:"keyspace_name"`
+	Table            string `json:"table_name"`
+	SubjectType      string `json:"subject_type"`
+	ErrorDescription string `json:"error_description"`
+	DocsUrl          string `json:"docs_url"`
+}
+
+type TestDataImportSourceResponse struct {
+	CanConnect   bool                              `json:"can_connect"`
+	ConnectError string                            `json:"error"`
+	Errors       []*DataSourceIncompatibilityError `json:"lint_errors"`
+}
+
+type StartDataImportRequest struct {
+	DatabaseName string           `json:"name"`
+	Organization string           `json:"organization"`
+	Source       DataImportSource `json:"connection"`
+}
+
+type MakePlanetScalePrimaryRequest struct {
+	Organization string `json:"organization"`
+	Database     string `json:"name"`
+}
+
+type MakePlanetScaleReplicaRequest struct {
+	Organization string `json:"organization"`
+	Database     string `json:"name"`
```

Comment:
Nit: For a separate PR and same as earlier, name here is a little ambiguous. Could we change this to be `database_name`?